### PR TITLE
Update EmailService to use Amazon SES instead of Mailgun

### PR DESCRIPTION
**Pull request by rymaju**
Note: I made a change to the prettier config so that it lines up with our frontend code. Turns out that made the git diff awful. Completely my fault. Just skip the files that aren't mentioned in the Changes section below.
---
# Summary
Worked with @jhylisa666 to do this!
Since we're moving away from Mailgun as an email service and using Amazon SES, we changed the NestJS modules in `/email` to use Amazon SES instead.

## Changes
- Added `amazon-ses.wrapper.ts` and `amazon-ses-client.factory.ts` and removed the Mailgun equivalents
- Replaced Mailgun module imports and provider imports with new Amazon SES equivalents.
- Added new environment variables to `.example.env`
  - `AWS_SES_REGION`
  - `AWS_SES_ACCESS_KEY`
  - `AWS_SES_ACCESS_KEY_SECRET`
  - `AWS_SES_SENDER_EMAIL`
- Removed `.env.example`

## Verification & Example

Configure `.env` to include the new environment variables like so:

```
AWS_SES_REGION=xxx
AWS_SES_ACCESS_KEY_ID=xxx
AWS_SES_SECRET_ACCESS_KEY=xxx
AWS_SES_SENDER_EMAIL=xxx
```

```ts
await service.sendEmails(['my.email@gmail.com], 'Test SES Email', '<h1>Test</h1>')
```

Results in an email like:
image

### CODE REVIEW

File: src/util/email/amazon-ses.wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      Source: '<email address you verified>',
+      Destination: {
+        ToAddresses: recipientEmails,
+      },
+      ReplyToAddresses: [],
+      Message: {
+        Body: {
+          Html: {
+            Charset: 'UTF-8',
+            Data: bodyHtml,
```
Comment (onb6) [inline src/util/email/amazon-ses.wrapper.ts:33]: Wait this is so beautifully simple

File: src/util/email/mailgun.wrapper.ts [hunk 2]
```diff
@@ -11,12 +11,7 @@ export class MailgunWrapper {
     this.client = client;
   }
 
-  sendEmails(
-    data:
-      | Mailgun.messages.SendData
-      | SendTemplateData
-      | Mailgun.messages.BatchData,
-  ) {
+  sendEmails(data: Mailgun.messages.SendData | SendTemplateData | Mailgun.messages.BatchData) {
```
Comment (onb6) [inline src/util/email/mailgun.wrapper.ts:14]: Why is this file still being used?

File: src/util/email/email.service.spec.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { Test, TestingModule } from '@nestjs/testing';
+import { AmazonSESWrapper } from './amazon-ses.wrapper';
 import { EmailService } from './email.service';
-import { MailgunWrapper } from './mailgun.wrapper';
 
-const mockMailgunWrapper = {
+const mockAmazonSESWrapper = {
```
Comment (onb6) [inline src/util/email/email.service.spec.ts:5]: Just wondering, have we been able to send an email successfully using this?

File: src/util/email/mailgun.wrapper.ts (hunk 2 above)
Comment (rymaju) [reply src/util/email/mailgun.wrapper.ts:14]: Excellent question. I'll wipe it from existence. 😈

File: src/util/email/email.service.spec.ts (hunk 3 above)
Comment (rymaju) [reply src/util/email/email.service.spec.ts:5]: We had some issues getting SES set up but I was able to get it working after sending an email and waiting overnight. We'll be able to test it later today.

File: src/util/email/amazon-ses.wrapper.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+   *
+   * @param recipientEmails the email addresses of the recipients
+   * @param subject the subject of the email
+   * @param bodyHtml the HTML body of the email
+   * @resolves if the email was sent successfully
+   * @rejects if the email was not sent successfully
+   */
+  async sendEmails(recipientEmails: string[], subject: string, bodyHtml: string): Promise<unknown> {
+    const params = {
+      Source: '<email address you verified>',
```
Comment (rymaju) [inline src/util/email/amazon-ses.wrapper.ts:24]: Oops forgot to do something about this 😓

Comment (rymaju) [reply src/util/email/amazon-ses.wrapper.ts]: Should be addressed now, added a new env var for this

**Comment by rymaju** [comment]
So the git history is so mushed up that I cant just revert that one prettier change, sorry! 😔

File: src/util/email/email.service.spec.ts (hunk 3 above)
Comment (rymaju) [reply src/util/email/email.service.spec.ts:5]: Update: it works!

Comment (onb6) [reply src/util/email/email.service.spec.ts:5]: WOOHOOOOO

**Review by onb6 — APPROVED** [review]
Glad to hear it worked! LGTM!

**Pull request merged**